Anagram.java

```diff
@@ -24,26 +26,79 @@ public static void main(String args[]) {
 			if (!pass) break;
 		}
 		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+	}
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
-	}
+		String main_str = preProcess(str1);
+		String secondary_str = preProcess(str2);
+
+		for (int i = 0; i < main_str.length(); i++){
+			for (int j = 0; j < secondary_str.length(); j++){
+				if (main_str.charAt(i) == secondary_str.charAt(j)){
+
+					secondary_str = secondary_str.replaceFirst(Character.toString(secondary_str.charAt(j)), "");
+
+				}
+			}
+
+		}
+
+		for (int a = 0; a < secondary_str.length(); a++){
+			if (secondary_str.charAt(a) != ' '){
+				return false;
+			}
+		}
+		return true;
+
+		}
 	   
 	// Returns a preprocessed version of the given string: all the letter characters are converted
 	// to lower-case, and all the other characters are deleted, except for spaces, which are left
 	// as is. For example, the string "What? No way!" becomes "whatnoway"
+
 	public static String preProcess(String str) {
-		// Replace the following statement with your code
-		return "";
+		String preProccessedString = "";
+		char test_char;
```

Comment:
dont declare variables without initial value.. 
In Java, it can lead to exceptions